### PR TITLE
tests: convert thrift tests to use the new ztest api

### LINKED ISSUE #131: tests: convert thrift tests to use the new ztest api
**Is your enhancement proposal related to a problem? Please describe.**
The new ztest API is a bit more like gtest. In fact, it was added by Google and is being actively driven by a few Google employees.

Switching over to the new ztest API is relatively painless (although some slightly more advanced cases occur when there is a [test fixture](https://docs.zephyrproject.org/latest/develop/test/ztest.html#test-fixtures)).

```cpp
ZTEST(suite_name, test_foo)
{
...
}

ZTEST_SUITE(suite_name, predicate_obj, setup_fn, before_fn, teardown_fn, after_fn);
```

**Describe the solution you'd like**
convert suites under `tests/` to use the new ztest API

**Describe alternatives you've considered**
While it isn't the default (yet), it will likely be the default very soon. Ensuring that we are up-to-date with upstream's best practices will help when the project is scheduled to be upstreamed in September.

**Additional context**
See also https://docs.zephyrproject.org/latest/develop/test/ztest.html

### PR DESCRIPTION
**Pull request by SdtElectronics**
* Use new ztest api in test for compression library
* Use new ztest api in ThriftTest

Fixes #131

### CODE REVIEW

File: tests/lib/thrift/ThriftTest/prj.conf [hunk 1]
```diff
@@ -18,6 +18,7 @@ CONFIG_THRIFT=y
 
 # Test dependencies
 CONFIG_ZTEST=y
+CONFIG_ZTEST_NEW_API=y=y
```
Comment (stephanosio) [inline tests/lib/thrift/ThriftTest/prj.conf:21]: typo?

File: tests/lib/thrift/ThriftTest/src/client.cpp [hunk 2]
```diff
@@ -21,22 +21,22 @@ using namespace std;
 
 static void init_Xtruct(Xtruct& s);
 
-void test_void(void) {
+ZTEST(thrift_test, test_void) {
```
Comment (stephanosio) [inline tests/lib/thrift/ThriftTest/src/client.cpp:24]: no need to suffix testsuite names with `_test`

File: tests/lib/thrift/ThriftTest/src/main.cpp [hunk 3]
```diff
@@ -95,7 +75,7 @@ static std::unique_ptr<TServer> setup_server() {
       new TSimpleServer(processor, serverTransport, transportFactory, protocolFactory));
 }
 
-static void setup(void) {
+static void thrift_test_before(void* unused) {
```
Comment (stephanosio) [inline tests/lib/thrift/ThriftTest/src/main.cpp:79]:
```
static void thrift_test_before(void* data) {
...
ARG_UNUSED(data);
```

File: tests/lib/thrift/ThriftTest/src/main.cpp [hunk 4]
```diff
@@ -125,7 +105,7 @@ static void setup(void) {
   zassert_equal(0, rv, "pthread_create failed: %d", rv);
 }
 
-static void teardown(void) {
+static void thrift_test_after(void* unused_) {
```
Comment (stephanosio) [inline tests/lib/thrift/ThriftTest/src/main.cpp:105]:
```
static void thrift_test_after(void* data) {
...
ARG_UNUSED(data);
```

File: tests/lib/thrift/muzic/src/integration.c [hunk 5]
```diff
@@ -28,7 +28,7 @@ static const char sample[]
 static z_stream* deflater;
 static z_stream* inflater;
 
-static void setup() {
+static void muzic_test_before(void* unused) {
```
Comment (stephanosio) [inline tests/lib/thrift/muzic/src/integration.c:31]: ditto

File: tests/lib/thrift/muzic/src/integration.c [hunk 6]
```diff
@@ -58,7 +58,7 @@ static void setup() {
   memset(infl_buf, 0, sizeof(infl_buf));
 }
 
-static void teardown() {
+static void muzic_test_after(void* unused) {
```
Comment (stephanosio) [inline tests/lib/thrift/muzic/src/integration.c:61]: ditto

**Comment by cfriedt** [comment]
@SdtElectronics - could we possibly just use TLS sockets only for the TLS demo? The socket pair test is actually more useful in this case, because it allows us to test on all platforms (not just platforms that support networking).

**Comment by cfriedt** [comment]
The ZTEST commit has a typo in prj.conf that should be fixed.

**Review by cfriedt — APPROVED** [review]

**Pull request merged**